src/ring_theory/polynomial/content.lean

```diff
@@ -423,5 +437,122 @@ gcd_monoid_of_exists_lcm $ λ p q, begin
+end
+
+end
+
+section fraction_map
+variables {K : Type*} [field K] (f : fraction_map R K)
+
+lemma is_primitive.is_unit_iff_is_unit_map {p : polynomial R} (hp : p.is_primitive) :
+  is_unit p ↔ is_unit (p.map f.to_map) :=
+hp.is_unit_iff_is_unit_map_of_injective f.injective
+
+open localization_map
+
+lemma is_unit_or_eq_zero_of_is_unit_integer_normalization_prim_part
+  {p : polynomial K} (h0 : p ≠ 0) (h : is_unit (@integer_normalization _ _ _ _ _ f p).prim_part) :
+  is_unit p :=
+begin
+  rcases is_unit_iff.1 h with ⟨_, ⟨u, rfl⟩, hu⟩,
+  obtain ⟨⟨c, c0⟩, hc⟩ := @integer_normalization_map_to_map _ _ _ _ _ f p,
+  rw [algebra.smul_def, ← C_eq_algebra_map, subtype.coe_mk] at hc,
+  apply is_unit_of_mul_is_unit_right,
+  rw [← hc, (@integer_normalization _ _ _ _ _ f p).eq_C_content_mul_prim_part, ← hu,
+    ← ring_hom.map_mul, is_unit_iff],
+  refine ⟨f.to_map ((@integer_normalization _ _ _ _ _ f p).content * ↑u),
+    is_unit_iff_ne_zero.2 (λ con, _), by simp⟩,
+  replace con := (ring_hom.injective_iff f.to_map).1 f.injective _ con,
+  rw [mul_eq_zero, content_eq_zero_iff, fraction_map.integer_normalization_eq_zero_iff] at con,
+  rcases con with con | con,
+  { apply h0 con },
+  { apply units.ne_zero _ con },
+end
+
+/-- Gauss's Lemma states that a primitive polynomial is irreducible iff it is irreducible in the
+  fraction field. -/
+theorem is_primitive.irreducible_iff_irreducible_map_fraction_map
+  {p : polynomial R} (hp : p.is_primitive) :
+  irreducible p ↔ irreducible (p.map f.to_map) :=
+begin
+  refine ⟨λ hi, ⟨λ h, hi.1 ((hp.is_unit_iff_is_unit_map f).2 h), λ a b hab, _⟩,
+    hp.irreducible_of_irreducible_map_of_injective f.injective⟩,
+  obtain ⟨⟨c, c0⟩, hc⟩ := @integer_normalization_map_to_map _ _ _ _ _ f a,
+  obtain ⟨⟨d, d0⟩, hd⟩ := @integer_normalization_map_to_map _ _ _ _ _ f b,
+  rw [algebra.smul_def, ← C_eq_algebra_map, subtype.coe_mk] at hc hd,
+  rw [submonoid.mem_carrier, mem_non_zero_divisors_iff_ne_zero] at c0 d0,
+  have hcd0 : c * d ≠ 0 := mul_ne_zero c0 d0,
+  rw [ne.def, ← C_eq_zero] at hcd0,
+  have h1 : C c * C d * p = @integer_normalization _ _ _ _ _ f a
+    * @integer_normalization _ _ _ _ _ f b,
+  { apply (map_injective _ f.injective _),
+    rw [map_mul, map_mul, map_mul, hc, hd, map_C, map_C, hab],
+    ring },
+  obtain ⟨u, hu⟩ : associated (c * d) (content (@integer_normalization _ _ _ _ _ f a) *
+            content (@integer_normalization _ _ _ _ _ f b)),
+  { rw [← dvd_dvd_iff_associated, ← normalize_eq_normalize_iff, monoid_hom.map_mul,
+        monoid_hom.map_mul, normalize_content, normalize_content,
+        ← mul_one (normalize c * normalize d), ← hp.content_eq_one, ← content_C, ← content_C,
+        ← content_mul, ← content_mul, ← content_mul, h1] },
+  rw [← ring_hom.map_mul, eq_comm,
+    (@integer_normalization _ _ _ _ _ f a).eq_C_content_mul_prim_part,
+    (@integer_normalization _ _ _ _ _ f b).eq_C_content_mul_prim_part, mul_assoc,
```

Comment:
OK so for some reason I'm getting the feeling that the current decision about which variables in `integer_normalization` are implicit, is perhaps not the right one. But when I look at the definition it seems OK :-/ Sometimes there's a trick where you can say `(integer_normalization a : polynomial R)` or whatever type it should have -- but I didn't have much luck getting it working here.

Reply:
I've gone ahead and made f explicit for that function -- I haven't changed `integer_normalization_map_to_map` though, which I also used @ with.